### PR TITLE
Class table links are now real links

### PR DESCRIPTION
**Pull request by TheGiddyLimit**
In addition:
 - better support for sub-hashes
 - class features can now be directly linked

### CODE REVIEW

File: js/page-classes.js [hunk 1]
```diff
@@ -218,9 +220,10 @@ function loadhash (id) {
 			// write out list to class table
 			var multifeature = "";
 			if (curlevel.feature.length !== 1 && a !== 0) multifeature = ", ";
-			let featureSpan = document.createElement('span');
+			let featureSpan = document.createElement(ELE_A);
 			featureSpan.setAttribute('data-link', link);
```
Comment (aeosynth) [inline js/page-classes.js]: don't need `data-link` anymore

File: js/history.js [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
+	const $list = $("#listcontainer");
+	const $el = $list.find(`a[href='#${link.toLowerCase()}']`);
+	loadhash($el.attr("id"));
+	document.title = decodeURIComponent($el.attr("title")) + " - 5etools";
 
-	if (sub) {
-		loadsub(sub)
-	}
-}
+	loadsub(sub)
```
Comment (aeosynth) [inline js/history.js]: this will error on pages w/o `loadsub` defined

File: js/history.js [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+	const link = splitHash[0];
+	const sub = splitHash.slice(1);
 
-	if (!e || !sub) {
-		const $el = $(`#listcontainer a[href='#${link.toLowerCase()}']`);
-		loadhash($el.attr("id"));
-		document.title = decodeURIComponent($el.attr("title")) + " - 5etools";
-	}
+	const $list = $("#listcontainer");
+	const $el = $list.find(`a[href='#${link.toLowerCase()}']`);
```
Comment (aeosynth) [inline js/history.js]: just my pref, but there's really no reason to break this up into two lines

File: js/history.js [hunk 4]
```diff
@@ -1,13 +1,12 @@
 window.onhashchange = function hashchange(e) {
-	const [link, sub] = window.location.hash.slice(1).split(',');
+	const splitHash = window.location.hash.slice(1).split(',');
+	const link = splitHash[0];
+	const sub = splitHash.slice(1);
```
Comment (aeosynth) [inline js/history.js]: same here, no reason for this to be 3 lines instead of 1

Comment (aeosynth) [reply js/history.js]: `const [link, ...sub] = ...`

File: js/page-classes.js [hunk 5]
```diff
@@ -252,42 +255,111 @@ function loadhash (id) {
...
 	$("#subclasses > span").click(function() {
 		const name = $(this).children("span").text()
-		if ($(this).hasClass("active"))
-			window.location.hash = window.location.hash.replace(/\,.*/, "").toLowerCase()
-		else
-			window.location.hash = window.location.hash.replace(/\,.*|$/, "," + encodeURIComponent(name).replace("'", "%27")).toLowerCase()
+		window.location.hash = getSubclassedHash($(this), name)
+		$(".featurelink").each(
+			function() {
+				this.setAttribute(ATB_HREF, window.location.hash)
```
Comment (aeosynth) [inline js/page-classes.js]: what does this do?

File: js/history.js (hunk 4 above)
Comment (TheGiddyLimit) [reply js/history.js]: :ok_hand::ok_hand: that's some good stuff

**Comment by TheGiddyLimit** [comment]
I'll generalise it (or someone else can) when I/we rework Races to use subhashes

**Pull request merged**